e2etest/newe2e_sas_generation.go

```diff
@@ -88,6 +91,13 @@ func (vals GenericServiceSignatureValues) withDefaults() GenericServiceSignature
 func (vals GenericServiceSignatureValues) AsBlob() BlobSignatureValues {
 	s := vals.withDefaults()
 
+	blobName := s.ObjectName
+	directoryName := s.DirectoryPath
+	if s.Level != nil && *s.Level != cmd.ELocationLevel.Object() {
```

Comment:
Tying the level to the signature values smells to me. What happens if another container comes along wanting to use the same values, after an object did? The test writer would have to recall this functionality and account for it.

Reply:
updated to the new model we agreed on